crates/oxc_query/src/properties.rs

```diff
@@ -0,0 +1,584 @@
+                    };
+                    parts.push(static_membexpr.property.name.to_string());
+                    membexpr = expr.object();
+                }
+
+                let Expression::Identifier(ident) = membexpr else { unreachable!("TS:2499") };
+                parts.push(ident.name.to_string());
+
+                parts.reverse();
+
+                parts.join(".")
+            }
+        }
+        .into()
+    })
+}
+
+pub(super) fn resolve_assignment_type_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "assignment_to_variable_name" => resolve_property_with(contexts, |v| {
+            let Vertex::AssignmentType(BindingPatternKind::BindingIdentifier(ident)) = v else {
+                return FieldValue::Null;
+            };
+            ident.name.to_string().into()
+        }),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'AssignmentType'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_class_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "extended_class_name" => resolve_property_with(contexts, |v| {
+            let Some(Expression::Identifier(ref ident)) = v
+                .as_class()
+                .unwrap_or_else(|| panic!("expected to have a class vertex, instead have: {v:#?}"))
+                .class
+                .super_class
+            else {
+                return FieldValue::Null;
+            };
+            ident.name.to_string().into()
+        }),
+        "is_abstract" => resolve_property_with(contexts, |v| {
+            v.as_class()
+                .unwrap_or_else(|| panic!("expected to have a class vertex, instead have: {v:#?}"))
+                .class
+                .modifiers
+                .contains(oxc_ast::ast::ModifierKind::Abstract)
+                .into()
+        }),
+        _ => {
+            unreachable!("attempted to read unexpected property '{property_name}' on type 'Class'")
+        }
+    }
+}
+
+pub(super) fn resolve_class_method_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "accessibility" => resolve_property_with(contexts, |v| {
+            v.as_class_method()
+                .unwrap_or_else(|| {
+                    panic!("expected to have a classmethod vertex, instead have: {v:#?}")
+                })
+                .method
+                .accessibility
+                .map(accessibility_to_string)
```

Comment:
should `accessibility_to_string` return a FieldValue directly? since that's the only reason it's used...

Reply:
Probably a good idea. Possibly might need a rename as well.